### ISSUE TITLE
Use of uninitialized value in addition (+) at .../PDF/Table.pm Line 734

### ISSUE THREAD

**Issue by mdeweerd** [PhilterPaper/PDF-Table#71]
I get a fatal error due to an uninitialized value error on https://github.com/PhilterPaper/PDF-Table/blob/37b03b64e8c27849379c8fa61fdb6ee0f934265d/lib/PDF/Table.pm#L734, running on Cygwin.

I instrumented the code like this, I added parantheses to rule out doubts about && and || precedence:
```perl
for my $k (@{$rows_height}) { print STDERR "REF $k -".ref($k)."\n"; }
print STDERR "JOINED:".join(",",keys(@{$rows_height}))."\n";
#print STDERR "HEIGHT1:".$rows_height->[1]."\n";
print STDERR $do_headers;
print STDERR $pg_cnt;
print STDERR $min_height;
print STDERR "
        $min_height += rows_height->[1] if ($do_headers && $pg_cnt==1) ||
                                             ($do_headers==2 && $pg_cnt>1);
";
        $min_height += $rows_height->[1] if (($do_headers && $pg_cnt==1) ||
                                             ($do_headers==2 && $pg_cnt>1));

```

Which shows:
```
REF 14.0846456692913 -
JOINED:0
2114.0846456692913
        14.0846456692913 += rows_height->[1] if (2 && 1==1) ||
                                             (2==2 && 1>1);
Use of uninitialized value in addition (+) at /usr/local/share/perl5/site_perl/5.32/PDF/Table.pm line 744.
```

As "JOINED" is followed by 0, we known that $rows_height->[1] does not exist.

I presume that a better code is the following (line 734 in Table.pm), but I have not fully made out how $rows_height and $do_headers are set and used:
```perl
        $min_height += $rows_height->[1] if (($do_headers!=2 && $pg_cnt==1) ||
                                             ($do_headers==2 && $pg_cnt>1));
```

This may be because the table in particular had a header, but no content (I am not sure about that, I just got an unexpected empty table in my output).

**Comment by PhilterPaper** [comment]
Not that it probably matters here, but what are you expecting to see for `keys(@array)`? Apparently, according to https://perldoc.perl.org/functions/keys, that was experimental and has been withdrawn.

**Comment by mdeweerd** [comment]
Actually, I've known "keys" since I started using perl in 1998 or so.
It says "Since Perl 5.14, an experimental feature allowed keys to take a scalar expression", but it works on arrays/hashes to get the keys.

I used 'keys' to see if the index '1' was in the list of keys for $rows_height.  It was not when the exception occured, but it was for other cases.

**Comment by PhilterPaper** [comment]
The referenced link says:
```
Starting with Perl 5.14, an experimental feature allowed `keys` to take a scalar expression. This experiment has 
been deemed unsuccessful, and was removed as of Perl 5.24.
```
I'm not sure how this is supposed to behave with a non-scalar expressions. If I want to know how many elements are in an array, either `@array` in a scalar context, or `scalar(@array)` will tell me the top element number (+1).

**Comment by PhilterPaper** [comment]
Well, anyway, could you provide a small test case that shows this bug? Thank you.

**Comment by PhilterPaper** [comment]
I have reproduced this problem in both the official latest release of PDF::Table, and my in-progress working copy.

**Comment by mdeweerd** [comment]
Perfect ;-) .
I am mainly reporting to improve the module - my solution is to not have an empty table ;-), which requires some some effort to identify with a fatal error, so the fix can help some other developers or make more general cases work ;-).

**Comment by PhilterPaper** [comment]
The problem was code that didn't expect that there would be a header row with no data rows. This is fixed in the new release 1.003, which should be out within a few days.
